iele-examples/forwardingWallet-copycreate.iele

```diff
@@ -1,40 +1,75 @@
+// statically known but it is not dynamically generated either.
+
+// This contract implements a wallet that can serve deposit and withdrawal
+// requests as well as setup new accounts that forward funds sent to them to
+// this wallet. Only the owner can withdraw and/or setup new forwarders. The
+// contract deployed with a new forwarder account is found with (and copied
+// from) a provided account. An example of such a contract can be found in the
+// forwarder.iele example.
 contract Wallet {
-  // Only %owner can withdraw or create new forwarders
-  // Forwarder is taken from an existing contract on the blockchain. 
+
+  // initializes a wallet account by storing the provided account address in
+  // %owner as the wallet's owner and the provided account address in %forwarder
+  // as the account from where the forwarder contract can be copied.
   define @init(%owner, %forwarder) {
     %zero = 0
     %one = 1
     sstore %owner, %zero
     sstore %forwarder, %one
   }
  
+  // deposits the sent value to the wallet account's balance 
   define public @deposit() {}
 
+  // withdraws amount equal to %value from the wallet and sends it to the
+  // account address %to
   define public @withdraw(%to, %value) {
+    // get the wallet's owner from storage
     %zero = 0
     %owner = sload %zero
+
+    // ensure that the caller (the withdrawer) is the wallet's owner
     %caller = call @iele.caller()
     %isnotowner = cmp ne %caller, %owner
-    br %isnotowner, throw // throw if not authorized to withdraw from this account
+    br %isnotowner, throw
+
+    // withdraw the desired amount and send it to the desired account
     %gas = call @iele.gas()
-    %status = call @deposit at %to () send %value , gaslimit %gas // deposit funds in target account
+    %status = call @deposit at %to () send %value , gaslimit %gas
     br %status, throw // throw if call fails
     ret void
+
   throw:
     call @iele.invalid()
   }
 
+  // creates a new account that simply forwards any funds sent to it to this
+  // wallet and returns the address of the created forwarder account
   define public @newForwarder() {
+    // get the wallet's owner from storage
     %zero = 0
     %owner = sload %zero
+
+    // ensure that the caller is the wallet's owner
     %caller = call @iele.caller()
     %isnotowner = cmp ne %caller, %owner
-    br %isnotowner, throw // throw if not authorized to create new forwarder
+    br %isnotowner, throw
+
+    // get the address of the account with the forwarder contract to copy
     %one = 1
-    %forwarder = sload %one // load address of forwarder contract to copy
-    %status, %addr = copycreate %forwarder () send %zero // copy contract and re-initialize it
-    br %status, throw // throw if call fails
+    %forwarder = sload %one
+
+    // create a new account, deploy the forwarder contract copied from the
+    // designated account and return the new account's address
+    %status, %addr = copycreate %forwarder () send %zero
```

Comment:
Can you replace %zero and %one with immediates?

Reply:
done!